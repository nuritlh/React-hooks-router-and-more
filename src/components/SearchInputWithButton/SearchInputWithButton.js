import React, { useCallback, useState } from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import TextField from '@material-ui/core/TextField';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';
import ClearIcon from '@material-ui/icons/Clear';
import Typography from '@material-ui/core/Typography';

const styles = () => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    width: 250,
    border: `1px solid #d0cccc`,
    borderRadius: 4,
    backgroundColor: '#e7f0fe'
  },
  input: {
    flex: 1,
    marginLeft: 10
  },
  inputProps: {
    fontSize: 14,
    backgroundColor: '#e7f0fe'
  },
  divider: {
    height: 30,
    padding: 0.2
  },
  iconButton: {
    padding: 0,
    margin: "0 10px"
  },
  helpText: {
    padding: "4px 0"
  }
});

const SearchInputWithButton = ({ onSearch, placeholder, helpText, name, classes }) => {
  const [searchTerm, setSearchTerm] = useState('');

  const onChangeValue = useCallback((e) => {
    setSearchTerm(e.target.value);
  }, []);

  const onClearText = useCallback(() => {
    setSearchTerm('');
  },[]);

  const onSearchTerm = useCallback((val) => {
    onSearch(val);
  },[onSearch]);

  const onKeyPress = useCallback((e) => {
    if (e.which === 13) {
      e.preventDefault();
      onSearchTerm(e.target.value);
    }
  }, [onSearchTerm]);

  return (
    <>
      <div className={classes.root} id={`searchInput_${name}`}>
        <TextField
          placeholder={placeholder}
          className={classes.input}
          onChange={onChangeValue}
          value={searchTerm}
          onKeyDown={onKeyPress}
          size="small"
          name={`searchInput_${name}`}
          InputProps={{
            disableUnderline: true,
            classes: {
              input: classes.inputProps,
            }
          }}
        />
        { searchTerm !== '' &&
        <IconButton className={classes.iconButton} aria-label="clear" onClick={onClearText} name={`clearInputButton_${name}`}>
          <ClearIcon />
        </IconButton>
        }
        <Divider className={classes.divider} orientation="vertical" />
        <IconButton
          className={classes.iconButton}
          aria-label="search"
          onClick={() => onSearchTerm(searchTerm)}
          name={`searchInputButton_${name}`}
        >
          <SearchIcon />
        </IconButton>
      </div>
      <Typography variant="caption" display="block" gutterBottom color="textSecondary" className={classes.helpText}>
        {helpText}
      </Typography>
    </>
  );
};

SearchInputWithButton.defaultProps = {
  onSearch: () => {},
  placeholder: 'Search',
  helpText: ''
};

SearchInputWithButton.propTypes = {
  classes: PropTypes.object.isRequired,
  onSearch: PropTypes.func,
  placeholder: PropTypes.string,
  helpText: PropTypes.string,
  name: PropTypes.string.isRequired
};

export default withStyles(styles, { withTheme: true })(SearchInputWithButton);
