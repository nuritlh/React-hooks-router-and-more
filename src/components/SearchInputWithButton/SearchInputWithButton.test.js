import React from 'react';
import createMount from '@material-ui/core/test-utils/createMount';
import SearchInputWithButton from './SearchInputWithButton';

const mockProps = {
  onSearch: jest.fn(),
  placeholder: "Search",
  name: "searchInputWithButton",
  helpText: "Help text will appear here"
};

describe('should test SearchInputWithButton component', () => {
  let mount = createMount();
  let wrapper;
  let onClick;

  wrapper = mount(<SearchInputWithButton { ...mockProps} onClick={onClick}/>);

  it('should render the SearchInputWithButton Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render SearchInputWithButton button', () => {
    const searchInputWithButton = wrapper.find('#searchInput_searchInputWithButton');
    expect(searchInputWithButton.length).toEqual(1);
  });

  it('should render value and check clear button appears', () => {
    expect(wrapper.find('button[name="clearInputButton_searchInputWithButton"]').exists()).toBeFalsy();
    const SearchInput = wrapper.find('input[name="searchInput_searchInputWithButton"]');
    SearchInput.simulate('change', { target: { value: 'Hello' } });
    wrapper.update();
    expect(wrapper.find('button[name="clearInputButton_searchInputWithButton"]').exists()).toBeTruthy();
  });

  it('should check click on search button', () => {
    const SearchInputSearchButton = wrapper.find('button[name="searchInputButton_searchInputWithButton"]');
    SearchInputSearchButton.simulate('click');
    wrapper.update();
    expect(mockProps.onSearch).toBeCalled();
  });
});