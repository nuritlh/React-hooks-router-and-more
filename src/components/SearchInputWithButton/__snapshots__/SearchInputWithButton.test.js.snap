// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test SearchInputWithButton component should render the SearchInputWithButton Component correctly 1`] = `
<WithStyles(SearchInputWithButton)
  helpText="Help text will appear here"
  name="searchInputWithButton"
  onSearch={[MockFunction]}
  placeholder="Search"
>
  <SearchInputWithButton
    classes={
      Object {
        "divider": "SearchInputWithButton-divider-4",
        "helpText": "SearchInputWithButton-helpText-6",
        "iconButton": "SearchInputWithButton-iconButton-5",
        "input": "SearchInputWithButton-input-2",
        "inputProps": "SearchInputWithButton-inputProps-3",
        "root": "SearchInputWithButton-root-1",
      }
    }
    helpText="Help text will appear here"
    name="searchInputWithButton"
    onSearch={[MockFunction]}
    placeholder="Search"
    theme={
      Object {
        "breakpoints": Object {
          "between": [Function],
          "down": [Function],
          "keys": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
          "only": [Function],
          "up": [Function],
          "values": Object {
            "lg": 1280,
            "md": 960,
            "sm": 600,
            "xl": 1920,
            "xs": 0,
          },
          "width": [Function],
        },
        "direction": "ltr",
        "mixins": Object {
          "gutters": [Function],
          "toolbar": Object {
            "@media (min-width:0px) and (orientation: landscape)": Object {
              "minHeight": 48,
            },
            "@media (min-width:600px)": Object {
              "minHeight": 64,
            },
            "minHeight": 56,
          },
        },
        "overrides": Object {},
        "palette": Object {
          "action": Object {
            "activatedOpacity": 0.12,
            "active": "rgba(0, 0, 0, 0.54)",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "disabledBackground": "rgba(0, 0, 0, 0.12)",
            "disabledOpacity": 0.38,
            "focus": "rgba(0, 0, 0, 0.12)",
            "focusOpacity": 0.12,
            "hover": "rgba(0, 0, 0, 0.04)",
            "hoverOpacity": 0.04,
            "selected": "rgba(0, 0, 0, 0.08)",
            "selectedOpacity": 0.08,
          },
          "augmentColor": [Function],
          "background": Object {
            "default": "#fafafa",
            "paper": "#fff",
          },
          "common": Object {
            "black": "#000",
            "white": "#fff",
          },
          "contrastThreshold": 3,
          "divider": "rgba(0, 0, 0, 0.12)",
          "error": Object {
            "contrastText": "#fff",
            "dark": "#d32f2f",
            "light": "#e57373",
            "main": "#f44336",
          },
          "getContrastText": [Function],
          "grey": Object {
            "100": "#f5f5f5",
            "200": "#eeeeee",
            "300": "#e0e0e0",
            "400": "#bdbdbd",
            "50": "#fafafa",
            "500": "#9e9e9e",
            "600": "#757575",
            "700": "#616161",
            "800": "#424242",
            "900": "#212121",
            "A100": "#d5d5d5",
            "A200": "#aaaaaa",
            "A400": "#303030",
            "A700": "#616161",
          },
          "info": Object {
            "contrastText": "#fff",
            "dark": "#1976d2",
            "light": "#64b5f6",
            "main": "#2196f3",
          },
          "primary": Object {
            "contrastText": "#fff",
            "dark": "#303f9f",
            "light": "#7986cb",
            "main": "#3f51b5",
          },
          "secondary": Object {
            "contrastText": "#fff",
            "dark": "#c51162",
            "light": "#ff4081",
            "main": "#f50057",
          },
          "success": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "#388e3c",
            "light": "#81c784",
            "main": "#4caf50",
          },
          "text": Object {
            "disabled": "rgba(0, 0, 0, 0.38)",
            "hint": "rgba(0, 0, 0, 0.38)",
            "primary": "rgba(0, 0, 0, 0.87)",
            "secondary": "rgba(0, 0, 0, 0.54)",
          },
          "tonalOffset": 0.2,
          "type": "light",
          "warning": Object {
            "contrastText": "rgba(0, 0, 0, 0.87)",
            "dark": "#f57c00",
            "light": "#ffb74d",
            "main": "#ff9800",
          },
        },
        "props": Object {},
        "shadows": Array [
          "none",
          "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
          "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
          "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
          "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
          "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
          "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
          "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
          "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
          "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
          "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
          "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
          "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
          "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
          "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
          "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
          "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
          "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
          "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
          "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
          "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
          "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
        ],
        "shape": Object {
          "borderRadius": 4,
        },
        "spacing": [Function],
        "transitions": Object {
          "create": [Function],
          "duration": Object {
            "complex": 375,
            "enteringScreen": 225,
            "leavingScreen": 195,
            "short": 250,
            "shorter": 200,
            "shortest": 150,
            "standard": 300,
          },
          "easing": Object {
            "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
            "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
            "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
            "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
          },
          "getAutoHeightDuration": [Function],
        },
        "typography": Object {
          "body1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "letterSpacing": "0.00938em",
            "lineHeight": 1.5,
          },
          "body2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 400,
            "letterSpacing": "0.01071em",
            "lineHeight": 1.43,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "letterSpacing": "0.02857em",
            "lineHeight": 1.75,
            "textTransform": "uppercase",
          },
          "caption": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "letterSpacing": "0.03333em",
            "lineHeight": 1.66,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
          "fontSize": 14,
          "fontWeightBold": 700,
          "fontWeightLight": 300,
          "fontWeightMedium": 500,
          "fontWeightRegular": 400,
          "h1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "6rem",
            "fontWeight": 300,
            "letterSpacing": "-0.01562em",
            "lineHeight": 1.167,
          },
          "h2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3.75rem",
            "fontWeight": 300,
            "letterSpacing": "-0.00833em",
            "lineHeight": 1.2,
          },
          "h3": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "3rem",
            "fontWeight": 400,
            "letterSpacing": "0em",
            "lineHeight": 1.167,
          },
          "h4": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "2.125rem",
            "fontWeight": 400,
            "letterSpacing": "0.00735em",
            "lineHeight": 1.235,
          },
          "h5": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.5rem",
            "fontWeight": 400,
            "letterSpacing": "0em",
            "lineHeight": 1.334,
          },
          "h6": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1.25rem",
            "fontWeight": 500,
            "letterSpacing": "0.0075em",
            "lineHeight": 1.6,
          },
          "htmlFontSize": 16,
          "overline": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.75rem",
            "fontWeight": 400,
            "letterSpacing": "0.08333em",
            "lineHeight": 2.66,
            "textTransform": "uppercase",
          },
          "pxToRem": [Function],
          "round": [Function],
          "subtitle1": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "1rem",
            "fontWeight": 400,
            "letterSpacing": "0.00938em",
            "lineHeight": 1.75,
          },
          "subtitle2": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": "0.875rem",
            "fontWeight": 500,
            "letterSpacing": "0.00714em",
            "lineHeight": 1.57,
          },
        },
        "zIndex": Object {
          "appBar": 1100,
          "drawer": 1200,
          "mobileStepper": 1000,
          "modal": 1300,
          "snackbar": 1400,
          "speedDial": 1050,
          "tooltip": 1500,
        },
      }
    }
  >
    <div
      className="SearchInputWithButton-root-1"
      id="searchInput_searchInputWithButton"
    >
      <WithStyles(ForwardRef(TextField))
        InputProps={
          Object {
            "classes": Object {
              "input": "SearchInputWithButton-inputProps-3",
            },
            "disableUnderline": true,
          }
        }
        className="SearchInputWithButton-input-2"
        name="searchInput_searchInputWithButton"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        size="small"
        value=""
      >
        <ForwardRef(TextField)
          InputProps={
            Object {
              "classes": Object {
                "input": "SearchInputWithButton-inputProps-3",
              },
              "disableUnderline": true,
            }
          }
          className="SearchInputWithButton-input-2"
          classes={
            Object {
              "root": "MuiTextField-root",
            }
          }
          name="searchInput_searchInputWithButton"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search"
          size="small"
          value=""
        >
          <WithStyles(ForwardRef(FormControl))
            className="MuiTextField-root SearchInputWithButton-input-2"
            color="primary"
            disabled={false}
            error={false}
            fullWidth={false}
            onKeyDown={[Function]}
            required={false}
            size="small"
            variant="standard"
          >
            <ForwardRef(FormControl)
              className="MuiTextField-root SearchInputWithButton-input-2"
              classes={
                Object {
                  "fullWidth": "MuiFormControl-fullWidth",
                  "marginDense": "MuiFormControl-marginDense",
                  "marginNormal": "MuiFormControl-marginNormal",
                  "root": "MuiFormControl-root",
                }
              }
              color="primary"
              disabled={false}
              error={false}
              fullWidth={false}
              onKeyDown={[Function]}
              required={false}
              size="small"
              variant="standard"
            >
              <div
                className="MuiFormControl-root MuiTextField-root SearchInputWithButton-input-2"
                onKeyDown={[Function]}
              >
                <WithStyles(ForwardRef(Input))
                  autoFocus={false}
                  classes={
                    Object {
                      "input": "SearchInputWithButton-inputProps-3",
                    }
                  }
                  disableUnderline={true}
                  fullWidth={false}
                  multiline={false}
                  name="searchInput_searchInputWithButton"
                  onChange={[Function]}
                  placeholder="Search"
                  value=""
                >
                  <ForwardRef(Input)
                    autoFocus={false}
                    classes={
                      Object {
                        "colorSecondary": "MuiInput-colorSecondary",
                        "disabled": "Mui-disabled",
                        "error": "Mui-error",
                        "focused": "Mui-focused",
                        "formControl": "MuiInput-formControl",
                        "fullWidth": "MuiInput-fullWidth",
                        "input": "MuiInput-input SearchInputWithButton-inputProps-3",
                        "inputMarginDense": "MuiInput-inputMarginDense",
                        "inputMultiline": "MuiInput-inputMultiline",
                        "inputTypeSearch": "MuiInput-inputTypeSearch",
                        "marginDense": "MuiInput-marginDense",
                        "multiline": "MuiInput-multiline",
                        "root": "MuiInput-root",
                        "underline": "MuiInput-underline",
                      }
                    }
                    disableUnderline={true}
                    fullWidth={false}
                    multiline={false}
                    name="searchInput_searchInputWithButton"
                    onChange={[Function]}
                    placeholder="Search"
                    value=""
                  >
                    <WithStyles(ForwardRef(InputBase))
                      autoFocus={false}
                      classes={
                        Object {
                          "colorSecondary": "MuiInput-colorSecondary",
                          "disabled": "Mui-disabled",
                          "error": "Mui-error",
                          "focused": "Mui-focused",
                          "formControl": "MuiInput-formControl",
                          "fullWidth": "MuiInput-fullWidth",
                          "input": "MuiInput-input SearchInputWithButton-inputProps-3",
                          "inputMarginDense": "MuiInput-inputMarginDense",
                          "inputMultiline": "MuiInput-inputMultiline",
                          "inputTypeSearch": "MuiInput-inputTypeSearch",
                          "marginDense": "MuiInput-marginDense",
                          "multiline": "MuiInput-multiline",
                          "root": "MuiInput-root",
                          "underline": null,
                        }
                      }
                      fullWidth={false}
                      inputComponent="input"
                      multiline={false}
                      name="searchInput_searchInputWithButton"
                      onChange={[Function]}
                      placeholder="Search"
                      type="text"
                      value=""
                    >
                      <ForwardRef(InputBase)
                        autoFocus={false}
                        classes={
                          Object {
                            "adornedEnd": "MuiInputBase-adornedEnd",
                            "adornedStart": "MuiInputBase-adornedStart",
                            "colorSecondary": "MuiInputBase-colorSecondary MuiInput-colorSecondary",
                            "disabled": "Mui-disabled Mui-disabled",
                            "error": "Mui-error Mui-error",
                            "focused": "Mui-focused Mui-focused",
                            "formControl": "MuiInputBase-formControl MuiInput-formControl",
                            "fullWidth": "MuiInputBase-fullWidth MuiInput-fullWidth",
                            "input": "MuiInputBase-input MuiInput-input SearchInputWithButton-inputProps-3",
                            "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                            "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                            "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                            "inputMarginDense": "MuiInputBase-inputMarginDense MuiInput-inputMarginDense",
                            "inputMultiline": "MuiInputBase-inputMultiline MuiInput-inputMultiline",
                            "inputTypeSearch": "MuiInputBase-inputTypeSearch MuiInput-inputTypeSearch",
                            "marginDense": "MuiInputBase-marginDense MuiInput-marginDense",
                            "multiline": "MuiInputBase-multiline MuiInput-multiline",
                            "root": "MuiInputBase-root MuiInput-root",
                          }
                        }
                        fullWidth={false}
                        inputComponent="input"
                        multiline={false}
                        name="searchInput_searchInputWithButton"
                        onChange={[Function]}
                        placeholder="Search"
                        type="text"
                        value=""
                      >
                        <div
                          className="MuiInputBase-root MuiInput-root MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                          onClick={[Function]}
                        >
                          <input
                            aria-invalid={false}
                            autoFocus={false}
                            className="MuiInputBase-input MuiInput-input SearchInputWithButton-inputProps-3 MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                            disabled={false}
                            name="searchInput_searchInputWithButton"
                            onAnimationStart={[Function]}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            placeholder="Search"
                            required={false}
                            type="text"
                            value=""
                          />
                        </div>
                      </ForwardRef(InputBase)>
                    </WithStyles(ForwardRef(InputBase))>
                  </ForwardRef(Input)>
                </WithStyles(ForwardRef(Input))>
              </div>
            </ForwardRef(FormControl)>
          </WithStyles(ForwardRef(FormControl))>
        </ForwardRef(TextField)>
      </WithStyles(ForwardRef(TextField))>
      <WithStyles(ForwardRef(Divider))
        className="SearchInputWithButton-divider-4"
        orientation="vertical"
      >
        <ForwardRef(Divider)
          className="SearchInputWithButton-divider-4"
          classes={
            Object {
              "absolute": "MuiDivider-absolute",
              "flexItem": "MuiDivider-flexItem",
              "inset": "MuiDivider-inset",
              "light": "MuiDivider-light",
              "middle": "MuiDivider-middle",
              "root": "MuiDivider-root",
              "vertical": "MuiDivider-vertical",
            }
          }
          orientation="vertical"
        >
          <hr
            className="MuiDivider-root SearchInputWithButton-divider-4 MuiDivider-vertical"
          />
        </ForwardRef(Divider)>
      </WithStyles(ForwardRef(Divider))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="search"
        className="SearchInputWithButton-iconButton-5"
        name="searchInputButton_searchInputWithButton"
        onClick={[Function]}
      >
        <ForwardRef(IconButton)
          aria-label="search"
          className="SearchInputWithButton-iconButton-5"
          classes={
            Object {
              "colorInherit": "MuiIconButton-colorInherit",
              "colorPrimary": "MuiIconButton-colorPrimary",
              "colorSecondary": "MuiIconButton-colorSecondary",
              "disabled": "Mui-disabled",
              "edgeEnd": "MuiIconButton-edgeEnd",
              "edgeStart": "MuiIconButton-edgeStart",
              "label": "MuiIconButton-label",
              "root": "MuiIconButton-root",
              "sizeSmall": "MuiIconButton-sizeSmall",
            }
          }
          name="searchInputButton_searchInputWithButton"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(ButtonBase))
            aria-label="search"
            centerRipple={true}
            className="MuiIconButton-root SearchInputWithButton-iconButton-5"
            disabled={false}
            focusRipple={true}
            name="searchInputButton_searchInputWithButton"
            onClick={[Function]}
          >
            <ForwardRef(ButtonBase)
              aria-label="search"
              centerRipple={true}
              className="MuiIconButton-root SearchInputWithButton-iconButton-5"
              classes={
                Object {
                  "disabled": "Mui-disabled",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiButtonBase-root",
                }
              }
              disabled={false}
              focusRipple={true}
              name="searchInputButton_searchInputWithButton"
              onClick={[Function]}
            >
              <button
                aria-label="search"
                className="MuiButtonBase-root MuiIconButton-root SearchInputWithButton-iconButton-5"
                disabled={false}
                name="searchInputButton_searchInputWithButton"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <ForwardRef>
                    <WithStyles(ForwardRef(SvgIcon))>
                      <ForwardRef(SvgIcon)
                        classes={
                          Object {
                            "colorAction": "MuiSvgIcon-colorAction",
                            "colorDisabled": "MuiSvgIcon-colorDisabled",
                            "colorError": "MuiSvgIcon-colorError",
                            "colorPrimary": "MuiSvgIcon-colorPrimary",
                            "colorSecondary": "MuiSvgIcon-colorSecondary",
                            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                            "root": "MuiSvgIcon-root",
                          }
                        }
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </ForwardRef(SvgIcon)>
                    </WithStyles(ForwardRef(SvgIcon))>
                  </ForwardRef>
                </span>
                <WithStyles(memo)
                  center={true}
                >
                  <ForwardRef(TouchRipple)
                    center={true}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child",
                        "childLeaving": "MuiTouchRipple-childLeaving",
                        "childPulsate": "MuiTouchRipple-childPulsate",
                        "ripple": "MuiTouchRipple-ripple",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                        "root": "MuiTouchRipple-root",
                      }
                    }
                  >
                    <span
                      className="MuiTouchRipple-root"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component={null}
                        exit={true}
                      />
                    </span>
                  </ForwardRef(TouchRipple)>
                </WithStyles(memo)>
              </button>
            </ForwardRef(ButtonBase)>
          </WithStyles(ForwardRef(ButtonBase))>
        </ForwardRef(IconButton)>
      </WithStyles(ForwardRef(IconButton))>
    </div>
    <WithStyles(ForwardRef(Typography))
      className="SearchInputWithButton-helpText-6"
      color="textSecondary"
      display="block"
      gutterBottom={true}
      variant="caption"
    >
      <ForwardRef(Typography)
        className="SearchInputWithButton-helpText-6"
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        color="textSecondary"
        display="block"
        gutterBottom={true}
        variant="caption"
      >
        <span
          className="MuiTypography-root SearchInputWithButton-helpText-6 MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          Help text will appear here
        </span>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
  </SearchInputWithButton>
</WithStyles(SearchInputWithButton)>
`;
